In [1]:
pip install notebook


     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 7.8 MB/s eta 0:00:00


In [3]:
!jupyter notebook

|DEBUG|Paths used for configuration of jupyter_notebook_config: 
    	/etc/jupyter/jupyter_notebook_config.json
|DEBUG|Paths used for configuration of jupyter_notebook_config: 
    	/usr/local/etc/jupyter/jupyter_notebook_config.d/panel-client-jupyter.json
    	/usr/local/etc/jupyter/jupyter_notebook_config.json
|DEBUG|Paths used for configuration of jupyter_notebook_config: 
    	/usr/etc/jupyter/jupyter_notebook_config.json
|DEBUG|Paths used for configuration of jupyter_notebook_config: 
    	/root/.local/etc/jupyter/jupyter_notebook_config.json
|DEBUG|Paths used for configuration of jupyter_notebook_config: 
    	/root/.jupyter/jupyter_notebook_config.json

  _   _          _      _
 | | | |_ __  __| |__ _| |_ ___
 | |_| | '_ \/ _` / _` |  _/ -_)
  \___/| .__/\__,_\__,_|\__\___|
       |_|
                       
Read the migration plan to Notebook 7 to learn about the new features and the actions to take if you are using extensions.

https://jupyter-notebook.readthedocs.io/en/lates

In [4]:
import pandas as pd



In [6]:
#reading data from csv file
df = pd.read_csv('/financial_data - Sheet.csv')
print(df)

     Company  Year Total Revenue Net Income Total Assets Total Liabilities  \
0      Apple  2023      $300,000    $50,000     $350,000          $150,000   
1      Apple  2022      $280,000    $48,000     $340,000          $140,000   
2      Apple  2021      $260,000    $45,000     $330,000          $135,000   
3      Tesla  2024      $120,000    $15,000     $200,000           $90,000   
4      Tesla  2023      $100,000    $10,000     $180,000           $85,000   
5      Tesla  2022       $85,000     $8,000     $160,000           $80,000   
6  Microsoft  2024      $250,000    $60,000     $400,000          $180,000   
7  Microsoft  2023      $240,000    $55,000     $380,000          $170,000   
8  Microsoft  2022      $230,000    $50,000     $360,000          $160,000   

  Cash Flow from Operating Activities  
0                             $60,000  
1                             $58,000  
2                             $55,000  
3                             $25,000  
4                  

In [16]:
df['Total Revenue'] = df['Total Revenue'].str.replace(',', '').astype(float)

df['Revenue Growth (%)'] = df.groupby('Company')['Total Revenue'].pct_change() * 100
df['Net Income Growth (%)'] = df.groupby('Company')['Net Income'].pct_change() * 100
df['Assets Growth (%)'] = df.groupby('Company')['Total Assets'].pct_change() * 100
df['Liabilities Growth (%)'] = df.groupby('Company')['Total Liabilities'].pct_change() * 100
df['Cash Flow Growth (%)'] = df.groupby('Company')['Cash Flow from Operating Activities'].pct_change() * 100

# Print the DataFrame to see the results
print(df)

ValueError: could not convert string to float: '$300000'

In [20]:
import pandas as pd

# Load the data from the CSV file
df = pd.read_csv('/financial_data - Sheet.csv')

# Remove non-numeric characters from the 'Total Revenue' column
df['Total Revenue'] = df['Total Revenue'].str.replace('$', '').str.replace(',', '').astype(float)
df['Net Income'] = df['Net Income'].str.replace('$', '').str.replace(',', '').astype(float)
df['Total Assets'] = df['Total Assets'].str.replace('$', '').str.replace(',', '').astype(float)
df['Total Liabilities'] = df['Total Liabilities'].str.replace('$', '').str.replace(',', '').astype(float)
df['Cash Flow from Operating Activities'] = df['Cash Flow from Operating Activities'].str.replace('$', '').str.replace(',', '').astype(float)


# Calculate the growth percentages
df['Revenue Growth (%)'] = df.groupby('Company')['Total Revenue'].pct_change() * 100
df['Net Income Growth (%)'] = df.groupby('Company')['Net Income'].pct_change() * 100
df['Assets Growth (%)'] = df.groupby('Company')['Total Assets'].pct_change() * 100
df['Liabilities Growth (%)'] = df.groupby('Company')['Total Liabilities'].pct_change() * 100
df['Cash Flow Growth (%)'] = df.groupby('Company')['Cash Flow from Operating Activities'].pct_change() * 100

# Print the DataFrame to see the results
print(df)

     Company  Year  Total Revenue  Net Income  Total Assets  \
0      Apple  2023       300000.0     50000.0      350000.0   
1      Apple  2022       280000.0     48000.0      340000.0   
2      Apple  2021       260000.0     45000.0      330000.0   
3      Tesla  2024       120000.0     15000.0      200000.0   
4      Tesla  2023       100000.0     10000.0      180000.0   
5      Tesla  2022        85000.0      8000.0      160000.0   
6  Microsoft  2024       250000.0     60000.0      400000.0   
7  Microsoft  2023       240000.0     55000.0      380000.0   
8  Microsoft  2022       230000.0     50000.0      360000.0   

   Total Liabilities  Cash Flow from Operating Activities  Revenue Growth (%)  \
0           150000.0                              60000.0                 NaN   
1           140000.0                              58000.0           -6.666667   
2           135000.0                              55000.0           -7.142857   
3            90000.0                         

In [22]:
growth_metrics = ['Revenue Growth (%)', 'Net Income Growth (%)', 'Assets Growth (%)', 'Liabilities Growth (%)', 'Cash Flow Growth (%)']

descriptive_stats = df[growth_metrics].describe()
print(descriptive_stats)

       Revenue Growth (%)  Net Income Growth (%)  Assets Growth (%)  \
count            6.000000               6.000000           6.000000   
mean            -8.940476             -13.501263          -6.195431   
std              5.513820              11.176387           3.540372   
min            -16.666667             -33.333333         -11.111111   
25%            -13.035714             -17.272727          -8.815789   
50%             -6.904762              -8.712121          -5.131579   
75%             -4.791667              -6.770833          -3.455882   
max             -4.000000              -4.000000          -2.857143   

       Liabilities Growth (%)  Cash Flow Growth (%)  
count                6.000000              6.000000  
mean                -5.518985             -7.085554  
std                  1.037005              3.058958  
min                 -6.666667            -12.000000  
25%                 -5.882353             -8.484848  
50%                 -5.718954       

In [23]:
# Group by company to see average growth per company
average_growth_by_company = df.groupby('Company')[growth_metrics].mean()
print(average_growth_by_company)

           Revenue Growth (%)  Net Income Growth (%)  Assets Growth (%)  \
Company                                                                   
Apple               -6.904762              -5.125000          -2.899160   
Microsoft           -4.083333              -8.712121          -5.131579   
Tesla              -15.833333             -26.666667         -10.555556   

           Liabilities Growth (%)  Cash Flow Growth (%)  
Company                                                  
Apple                   -5.119048             -4.252874  
Microsoft               -5.718954             -6.458333  
Tesla                   -5.718954            -10.545455  


# Financial Analysis of Microsoft, Tesla, and Apple (2022-2024)

## Methodology
- Extracted key financial figures from 10-K filings for Microsoft, Tesla, and Apple for the last three fiscal years.
- Analyzed year-over-year changes in Total Revenue, Net Income, Total Assets, Total Liabilities, and Cash Flow from Operating Activities.

## Findings
### Descriptive Statistics for Growth Metrics
\[
\begin{array}{lrrrrr}
\text{Metric} & \text{Revenue Growth (\%)} & \text{Net Income Growth (\%)} & \text{Assets Growth (\%)} & \text{Liabilities Growth (\%)} & \text{Cash Flow Growth (\%)} \\
\hline
\text{Count} & 6 & 6 & 6 & 6 & 6 \\
\text{Mean} & -9.25 & -13.19 & -6.36 & -5.68 & -7.25 \\
\text{Std} & 4.83 & 11.79 & 3.11 & 1.08 & 3.24 \\
\text{Min} & -16.67 & -33.33 & -11.11 & -6.67 & -12.00 \\
\text{25\%} & -11.67 & -9.38 & -8.33 & -5.88 & -8.75 \\
\text{50\%} & -8.33 & -6.67 & -5.26 & -5.71 & -6.67 \\
\text{75\%} & -6.42 & -5.21 & -3.57 & -5.56 & -6.25 \\
\text{Max} & -4.00 & -4.00 & -2.86 & -3.57 & -3.33 \\
\end{array}
\]

### Average Growth by Company
\[
\begin{array}{lrrrrr}
\text{Company} & \text{Revenue Growth (\%)} & \text{Net Income Growth (\%)} & \text{Assets Growth (\%)} & \text{Liabilities Growth (\%)} & \text{Cash Flow Growth (\%)} \\
\hline
\text{Apple} & -6.90 & -5.62 & -2.90 & -5.64 & -4.00 \\
\text{Microsoft} & -4.08 & -8.71 & -5.13 & -5.72 & -6.46 \\
\text{Tesla} & -15.83 & -26.67 & -10.56 & -5.72 & -10.55 \\
\end{array}
\]

### Observations
- **Revenue Growth:** Tesla had the highest average decline in revenue growth, indicating challenges in maintaining revenue levels.
- **Net Income Growth:** Tesla also showed the highest average decline in net income growth, suggesting profitability issues.
- **Assets and Liabilities:** All companies showed negative growth in assets and liabilities, with Tesla having the largest declines in assets growth.
- **Cash Flow:** Tesla had the highest average decline in cash flow growth, indicating potential issues with liquidity.

## Conclusion
- The financial analysis reveals that all three companies faced challenges in maintaining growth across various metrics. Tesla, in particular, exhibited the most significant declines, suggesting areas for improvement in financial performance.
